lazy_dataset/core.py

```diff
@@ -2763,6 +2758,88 @@ def __getitem__(self, item):
             return super().__getitem__(item)
 
 
+class TileDataset(Dataset):
+    """
+    Iterates over all elements of the input_dataset for `repetitions` times.
+
+    """
+
+    def __init__(self, input_dataset, repetitions):
+        """
+
+        Args:
+            input_dataset: dataset
+            repetitions: int
+
+        """
+        self.input_dataset = input_dataset
+        self.repetitions = repetitions
+
+    def copy(self, freeze=False):
+        return self.__class__(self.input_dataset.copy(freeze=freeze), self.repetitions)
+
+    @property
+    def indexable(self):
+        return self.input_dataset.indexable
+
+    @property
+    def ordered(self) -> bool:
+        return self.input_dataset.ordered
+
+    def __str__(self):
+        return f"{self.__class__.__name__}(repetitions={self.repetitions})"
+
+    def __iter__(self, with_key=False):
+        for _ in range(self.repetitions):
+            if with_key:
+                iterable = self.input_dataset.__iter__(with_key=True)
+            else:
+                iterable = self.input_dataset
+            for example in iterable:
+                yield example
+
+    def __len__(self):
+        return self.repetitions * len(self.input_dataset)
+
+    def __getitem__(self, item):
+        """
+        >>> ds = DictDataset({'a': {}, 'b': {}})
+        >>> ds = ds.items().map(lambda x: {'example_id': x[0], **x[1]})
+        >>> ds = ds.tile(2)
+        >>> len(ds)
+        4
+        >>> ds['a']
+        {'example_id': 'a'}
+        >>> ds['b']
+        {'example_id': 'b'}
+        >>> ds[5]
+        Traceback (most recent call last):
+          ...
+        IndexError: 5
+        >>> ds[-1]
+        {'example_id': 'b'}
+        >>> ds[-5]
+        Traceback (most recent call last):
+          ...
+        IndexError: -5
+
+        """
+        if isinstance(item, str):
+            return self.input_dataset[item]
+        elif isinstance(item, numbers.Integral):
```

Comment:
I am not sure, whether this code will have an effect in the performance.

How about changing the code to the follwing?
```python
input_len = len(self.input_dataset)
if not (-self.repetitions <= item // input_len < self.repetitions):
    raise IndexError(_item)
return self.input_dataset[item % input_len]
```